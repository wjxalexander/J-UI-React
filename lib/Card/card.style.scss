.j-ui-card-container{
  background: #fff;
  padding: 28px;
  margin-bottom: 28px;
  border-radius: 2px;
  background-clip: padding-box;
  box-shadow: 0 1.6px 3.6px 0 rgba(0,0,0,0.132), 0 0.3px 0.9px 0 rgba(0,0,0,0.108);
  animation: fade-in .8s cubic-bezier(0.1, 0.9, 0.2, 1) forwards;
  transform: translateY(1.2em);
 >h2{
   margin-bottom: 20px;
   color: #323130;
   font-weight: 600;
   line-height: 1;
   margin-top: 0;
 }
}
@keyframes fade-in {
  100% {
    opacity: 1;
    transform: none;
  }
}