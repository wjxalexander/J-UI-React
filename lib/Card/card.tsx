import React, {HTMLAttributes} from 'react';
import "./card.style.scss"
import {rootClass} from "../utils/classFactory";
import cls from "classnames";
interface CardProps extends HTMLAttributes<HTMLElement>{
  title?:string;
  className?:string
}
// HIGH ORDER FUNCTION
const classGenerator = rootClass('j-ui-card');
const Card:React.FunctionComponent<CardProps> = (props:CardProps) => {
  return (
    <div className={cls(classGenerator("container"),props.className)}>
      <h2>{props.title}</h2>
      {props.children}
    </div>
  );
};

export default Card;