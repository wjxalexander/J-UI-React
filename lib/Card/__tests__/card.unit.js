import * as React from "react"
import {shallow,mount} from 'enzyme';
import Card from"../card";

describe("card",()=>{
  it('card snapshot',()=>{
    const wrapper = shallow(<Card title={"test"}><div>test</div></Card>)
    expect(wrapper).toMatchSnapshot()
  })
});