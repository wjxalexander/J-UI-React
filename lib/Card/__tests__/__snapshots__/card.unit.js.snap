// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`card card snapshot 1`] = `
<div
  className="j-ui-card-container"
>
  <h2>
    test
  </h2>
  <div>
    test
  </div>
</div>
`;
